packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -189,6 +197,47 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
         emit FlashWBurn(msg.sender, _vaultId, _wPowerPerpAmountToBurn, _collateralToWithdraw, _wPowerPerpAmountToBuy);
     }
 
+    /**
+     * @notice sell long wPowerPerp and flashswap mint short position
+     * @dev flahswap amount = collateral amount - msg.value - ETH from selling long wPowerPerp
+     * @param _vaultId vault ID
+     * @param _wPowerPerpAmountToMint wPowerPerp amount to mint
+     * @param _collateralAmount collateral amount to use for minting
+     * @param _wPowerPerpAmountToSell long wPowerPerp amount to sell
+     * @param _minToReceive min ETH amount to receive for selling long wPowerPerp amount
+     */
+    function flashswapSellLongWMint(
+        uint256 _vaultId,
```

Comment:
Would it make sense to include existing vault collateral too to _collateralAmount is final amount in the vault?

Reply:
What we need the final vault collateral amount for ? `_collateralAmount` is for the amount to use for minting.

Reply:
are you thinking that someone may want to sell and withdraw collateral from the vault at same time?  is that the use case you are thinking?

maybe I should list every user flow today and make sure we check them all off

gamma's operate() functionality is nice for these kinds of things

Reply:
or thinking that someone wants to sell long, mint short sell, and withdraw collateral